server/text_generation_server/models/__init__.py

```diff
@@ -302,6 +302,9 @@ def get_model(
         if quantize in ["awq", "exl2", "gptq", "marlin"]:
             # These quantizers only work with float16 params.
             dtype = torch.float16
+        elif quantize == "fp8":
+            # gemm kernels are fp8xfp8->bf16
+            dtype = torch.bfloat16
```

Comment:
@danieldk is this compatible with the marlin kernels?

Reply:
Supports both `float16` and `bfloat16`.

However do we want to set the default to this, since most models are `float16`? Is it needed for the fbgemm quantization kernel?

Reply:
Yes it's required for fbgemm.
I can add a method in `layers/fp8.py` to check whether we will use fbgemm and set the default appropriately.

Reply:
Added a check.